### PR TITLE
synchronize remaining registry accesses

### LINKED ISSUE #65: Could not pass event PlayerInteractEvent to XClaim v1.10.5
Im using stock SPIGOT. Tried 1.20.4 and 1.20.2 versions, both have this issue while someone has opened his claiming interface console gets spammed with these messages constantly.

[23:06:57] [Server thread/ERROR]: Could not pass event PlayerInteractEvent to XClaim v1.10.5
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:310) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:570) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at org.bukkit.craftbukkit.v1_20_R2.event.CraftEventFactory.callPlayerInteractEvent(CraftEventFactory.java:532) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.level.PlayerInteractManager.a(PlayerInteractManager.java:187) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:1484) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.network.protocol.game.PacketPlayInBlockDig.a(SourceFile:42) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.network.protocol.game.PacketPlayInBlockDig.a(SourceFile:8) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.network.protocol.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:32) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.TickTask.run(SourceFile:18) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.util.thread.IAsyncTaskHandler.d(SourceFile:156) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(SourceFile:23) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1133) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:1) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.util.thread.IAsyncTaskHandler.x(SourceFile:130) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.bg(MinecraftServer.java:1112) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1105) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.util.thread.IAsyncTaskHandler.bp(SourceFile:115) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.u_(MinecraftServer.java:1088) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1000) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at net.minecraft.server.MinecraftServer.lambda$0(MinecraftServer.java:298) ~[spigot-1.20.2-R0.1-SNAPSHOT.jar:3950-Spigot-2888990-d485cde]
        at java.lang.Thread.run(Thread.java:833) ~[?:?]
Caused by: java.util.ConcurrentModificationException
        at java.util.HashMap$HashIterator.nextNode(HashMap.java:1597) ~[?:?]
        at java.util.HashMap$KeyIterator.next(HashMap.java:1620) ~[?:?]
        at codes.wasabi.xclaim.api.Claim.addChunk(Claim.java:352) ~[?:?]
        at codes.wasabi.xclaim.gui.ChunkEditor$Events.onInteract(ChunkEditor.java:227) ~[?:?]
        at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104) ~[?:?]
        at java.lang.reflect.Method.invoke(Method.java:577) ~[?:?]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:306) ~[spigot-api-1.20.2-R0.1-SNAPSHOT.jar:?]
        ... 23 more

### PR DESCRIPTION
**Pull request by PierreSchwang**
Should fix #65 - tho I've not been able to reproduce it. Some registry accesses were not properly synchronized / locked either way.

### CODE REVIEW

**Comment by WasabiThumb** [comment]
The point of registryLock is to make a synchronizedSet unecessary. The synchronizedSet is a quick catch-all that is not particularly performant and makes many assumptions about your data. There is no problem with the registry being a normal HashMap as long as registryLock is used to control access to it (which apparently, I missed a few cases). The other changes are good

File: src/main/java/codes/wasabi/xclaim/api/Claim.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 public class Claim {
 
-    private static final Set<Claim> registry = new HashSet<>();
+    private static final Set<Claim> registry = Collections.synchronizedSet(new HashSet<>());
```
Comment (WasabiThumb) [inline src/main/java/codes/wasabi/xclaim/api/Claim.java:34]: Synchronizing the set should not be necessary

**Review by WasabiThumb — CHANGES_REQUESTED** [review]

**Comment by PierreSchwang** [comment]
> The point of registryLock is to make a synchronizedSet unecessary. The synchronizedSet is a quick catch-all that is not particularly performant and makes many assumptions about your data. There is no problem with the registry being a normal HashMap as long as registryLock is used to control access to it (which apparently, I missed a few cases). The other changes are good

I'm not fully sure if enhanced for-loops are compiled into the Iteratables as well, because the Javadocs (https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/util/Collections.html#synchronizedSet(java.util.Set)) state "It is imperative that the user manually synchronize on the returned collection when traversing it via Iterator [...]"

**Comment by WasabiThumb** [comment]
> > The point of registryLock is to make a synchronizedSet unecessary. The synchronizedSet is a quick catch-all that is not particularly performant and makes many assumptions about your data. There is no problem with the registry being a normal HashMap as long as registryLock is used to control access to it (which apparently, I missed a few cases). The other changes are good
> 
> I'm not fully sure if enhanced for-loops are compiled into the Iteratables as well, because the Javadocs (https://docs.oracle.com/en/java/javase/20/docs/api/java.base/java/util/Collections.html#synchronizedSet(java.util.Set)) state "It is imperative that the user manually synchronize on the returned collection when traversing it via Iterator [...]"

ReadLock#lock() _is_ synchronization. Just a matter of terminology.

**Pull request merged**